network/controller/bootstrap/bootstrap.go

```diff
@@ -294,18 +301,27 @@ func (bc *Bootstrapper) startBootstrap(address string) (*host.Host, error) {
 		return nil, errors.Wrapf(err, "Failed to get response to bootstrap request from address %s", address)
 	}
 	data := response.GetData().(*NodeBootstrapResponse)
-	if data.Code == Rejected {
+	switch data.Code {
+	case Rejected:
 		return nil, errors.New("Rejected: " + data.RejectReason)
-	}
-	if data.Code == Redirected {
+	case Redirected:
 		return bootstrap(data.RedirectHost, bc.options, bc.startBootstrap)
 	}
-	return response.GetSenderHost(), nil
+	return &network.BootstrapResult{
+		PulseNum:       data.PulseNum,
+		FirstPulseTime: data.FirstPulseTime,
+		Host:           response.GetSenderHost(),
+	}, nil
 }
 
 func (bc *Bootstrapper) processBootstrap(ctx context.Context, request network.Request) (network.Response, error) {
 	// TODO: redirect logic
-	return bc.transport.BuildResponse(request, &NodeBootstrapResponse{Code: Accepted}), nil
+	return bc.transport.BuildResponse(request, &NodeBootstrapResponse{
+		Code:           Accepted,
+		FirstPulseTime: time.Now().UTC().Unix(),
```

Comment:
first pulse time should be initialized on Bootstrapper.Start

Reply:
fixed